.ci/no-exception-test.sh

```diff
@@ -0,0 +1,230 @@
+  cd ../../../
+  mvn -e clean install -Pno-validations
+  cd .ci-temp/contribution/checkstyle-tester
+  export MAVEN_OPTS="-Xmx2048m"
+  groovy ./launch.groovy --listOfProjects projects-to-test-on.properties \
+     --config checks-nonjavadoc-error.xml --checkstyleVersion $CS_POM_VERSION
+  cd ../..
+  removeFolderWithProtectedFiles contribution
+  ;;
+
+spotbugs)
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  echo CS_version: $CS_POM_VERSION
+  mkdir -p .ci-temp/
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/contribution
+  cd contribution/checkstyle-tester
+  sed -i.'' 's/^guava/#guava/' projects-to-test-on.properties
+  sed -i.'' 's/#spotbugs/spotbugs/' projects-to-test-on.properties
+  cd ../../../
+  mvn -e clean install -Pno-validations
+  cd .ci-temp/contribution/checkstyle-tester
+  export MAVEN_OPTS="-Xmx2048m"
+  groovy ./launch.groovy --listOfProjects projects-to-test-on.properties \
+    --config checks-nonjavadoc-error.xml --checkstyleVersion $CS_POM_VERSION
+  cd ../..
+  removeFolderWithProtectedFiles contribution
+  ;;
+
+spring-framework)
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  echo CS_version: $CS_POM_VERSION
+  mkdir -p .ci-temp/
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/contribution
+  cd contribution/checkstyle-tester
+  sed -i.'' 's/^guava/#guava/' projects-to-test-on.properties
+  sed -i.'' 's/#spring-framework/spring-framework/' projects-to-test-on.properties
+  cd ../../../
+  mvn -e clean install -Pno-validations
+  cd .ci-temp/contribution/checkstyle-tester
+  export MAVEN_OPTS="-Xmx2048m"
+  groovy ./launch.groovy --listOfProjects projects-to-test-on.properties \
+    --config checks-nonjavadoc-error.xml --checkstyleVersion $CS_POM_VERSION
+  cd ../..
+  removeFolderWithProtectedFiles contribution
+  ;;
+
+hbase)
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  echo CS_version: $CS_POM_VERSION
+  mkdir -p .ci-temp/
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/contribution
+  cd contribution/checkstyle-tester
+  sed -i.'' 's/^guava/#guava/' projects-to-test-on.properties
+  sed -i.'' 's/#Hbase/Hbase/' projects-to-test-on.properties
+  cd ../../../
+  mvn -e clean install -Pno-validations
+  cd .ci-temp/contribution/checkstyle-tester
+  export MAVEN_OPTS="-Xmx2048m"
+  groovy ./launch.groovy --listOfProjects projects-to-test-on.properties \
+      --config checks-nonjavadoc-error.xml --checkstyleVersion $CS_POM_VERSION
+  cd ../..
+  removeFolderWithProtectedFiles contribution
+  ;;
+
+pmd-elasticsearch-lombok-ast)
```

Comment:
https://github.com/checkstyle/checkstyle/blob/797642aef9585ac695686b98b7268caccd332a20/wercker.yml#L289

it might be absolutely the same.
We should not have duplicates in CI. please remove from no-exception-test.sh  what we have in wercker.sh

Wercker was created in same style as you create semaphore now, I did a copy of commands because I was not sure about stability of wercker at that time and wanted ability to quickly restore executions in wercker, and I never came back to this. Lets not repeat my mistake.

update of wercker.yml to reuse no-exception-test.sh is separate task.

Reply:
done